File: src/test/java/boj/problems/step2/No1330Test.java
package boj.problems.step2;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class No1330Test {
    @DisplayName("두 수 비교하기 테스트")
    @ParameterizedTest
    @CsvSource(value = {"1 2:<", "10 2:>", "5 5:=="}, delimiter = ':')
    void test(String input, String expected) throws IOException {
        System.out.println("두 수 비교하기 : https://www.acmicpc.net/problem/1330");

        BufferedReader br_given = new BufferedReader(new StringReader(input));

        assertEquals(expected, No1330.solve(br_given));

        br_given.close();
    }
}

File: src/test/kotlin/boj/problems/step2/No1330Test.kt
package boj.problems.step2

import io.kotest.core.spec.style.StringSpec
import io.kotest.data.row
import io.kotest.matchers.shouldBe
import java.io.BufferedReader
import java.io.StringReader

class No1330Test : StringSpec({
    "두 수 비교하기 테스트" {
        println("두 수 비교하기 : https://www.acmicpc.net/problem/1330")

        // Define the test cases as a Table
        val testCases = listOf(
            row("1 2", "<"),
            row("10 2", ">"),
            row("5 5", "=="),
        )

        // Run each test case
        testCases.forEach { (given, expected) ->
            // Run the test
            val input = BufferedReader(StringReader(given))

            val actual = No1330.solve(input)

            // verify the result
            actual shouldBe expected
        }
    }
})
